File: Models/Vacancy.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MtpApp.Models
{
    public class Vacancy
    {
        public int Id { get; set; }

        [Required]
        [Display(Name = "Posting Date")]
        public DateTime PostingDate { get; set; }

        [Required]
        public DateTime Deadline { get; set; }

        [Required]
        [Display(Name = "Employer")]
        public int EmployerId { get; set; }
        public Employer Employers { get; set; }

        [Required]
        [Display(Name = "Job Position")]
        public int JobPositionId { get; set; }
        public JobPosition JobPositions { get; set; }

        [Required]
        [Display(Name = "Job Category")]
        public int JobCategoryId { get; set; }
        public JobCategory JobCategories { get; set; }

        [Required]
        [Display(Name = "Position Available")]
        public int PositionAvailable { get; set; }

        [Display(Name = "Contract Type")]
        [StringLength(255)]
        public string ContractType { get; set; }

        [StringLength(255)]
        public string Schedule { get; set; }

        [Required]
        public Double Salary { get; set; }

        [Required]
        public double Salarymax { get; set; }

        [StringLength(510)]
        public string Location { get; set; }

        public string Responsibilities { get; set; }

        public string Requirement { get; set; }

        public string ApplicationInformation { get; set; }

        [Required]
        public string Status { get; set; }

        public int ViewCount { get; set; }

        [StringLength(255)]
        public string Branch { get; set; }

        public Vacancy()
        {
            ViewCount = ViewCount + 1;
        }
    }
}

File: spring-backend/src/main/java/com/mtp/api/models/Vacancy.java
package com.mtp.api.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "Vacancies", indexes = {
    @Index(name = "idx_vacancy_status", columnList = "status"),
    @Index(name = "idx_vacancy_employer", columnList = "EmployerId"),
    @Index(name = "idx_vacancy_category", columnList = "JobCategoryId")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Vacancy {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    private LocalDateTime postingDate;

    @NotNull
    private LocalDateTime deadline;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "EmployerId")
    private Employer employer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "JobPositionId")
    private JobPosition jobPosition;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "JobCategoryId")
    private JobCategory jobCategory;

    @NotNull
    private int positionAvailable;

    @Size(max = 255)
    private String contractType;

    @Size(max = 255)
    private String schedule;

    @NotNull
    private Double salary;

    @NotNull
    private Double salarymax;

    @Size(max = 510)
    private String location;

    @Column(columnDefinition = "TEXT")
    private String responsibilities;

    @Column(columnDefinition = "TEXT")
    private String requirement;

    @Column(columnDefinition = "TEXT")
    private String applicationInformation;

    @NotNull
    private String status;

    private int viewCount = 0;

    @Size(max = 255)
    private String branch;

    @Size(max = 500)
    private String imageUrl;
}

